src/main/java/com/google/sps/algorithms/MatchRanking.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps.algorithms;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class MatchRanking {
+
+    /**
+    * Function takes collection of user's saved bios, all user bios, and new match bios and 
+    * returns a list of the new match bios in ranked order from highest to lowest score 
+    * (higher score = more likely that the user will select it). 
+    */
+    public List<String> query(Collection<String> savedMatchBios, Collection<String> allUserBios, Collection<String> newMatchBios) {
+        Map<String, Double> newMatchScores = scoreNewMatches(savedMatchBios, allUserBios, newMatchBios);
+        return sortBiosByScore(newMatchScores);
+    }
+
+    /**
+    * Function that takes the three bio collections and returns a map with each new user bio mapped to it's score.
+    */
+    private Map<String, Double> scoreNewMatches(Collection<String> savedMatchBios, Collection<String> allUserBios, Collection<String> newMatchBios) {
```

Comment:
Would this just be to hold the bio score and bio string? The only problem is that the bio score will change whenever users save a new match so it might make it more complicated to have to update the Bio class every time.

Reply:
I was thinking you could have a new object with just the bio string. However, it probably makes more sense to use the Match object rather than create a new class. Using the Match object could allow you to rank based on the other fields in addition to the user's bio. Feel free to leave this as is.

Reply:
Oh yeah, I was planning on switching it to take in User objects in my next PR, I just wanted it to work with strings first!